chef-utils/lib/chef-utils/internal.rb

```diff
@@ -65,13 +75,39 @@ def __env_path
       end
     end
 
+    # This should be set to a Train::FIXME instance.  You should wire this up to nil for not using a train transport connection.
+    #
+    # @return [Train::FIXME]
+    #
     # @api private
+    #
     def __transport_connection
+      # Software consumers MUST override this method with their own implementation.  The default behavior here is subject to change.
       return Chef.run_context.transport_connection if defined?(Chef) && Chef.respond_to?(:run_context) && Chef&.run_context&.transport_connection
```

Comment:
I initially removed this line to force consumers to mixin the DSL methods and then force them to override this method (similar to __log and __config below).  I backed that out because that broke `ChefUtils.systemd?`.  The comment is meant to encourage any external consumers to treat this like `raise NotImplementedError` so that they are not tightly coupled to the magic here.  The `@api private` means that this code can change and break at any time.